mycore-base/src/main/java/org/mycore/common/MCRSession.java

```diff
@@ -513,7 +514,7 @@ public void setUserInformation(MCRUserInformation userSystemAdapter) {
      * @param task thread witch will be executed after an commit
      */
     public void onCommit(Runnable task) {
-        this.onCommitTasks.get().add(task);
+        this.onCommitTasks.get().add(new MCRTransactionableRunnable(task));
```

Comment:
Shouldn't it use `this`session when running the task instead of creating a new session that belongs to nobody?